envtest/ibmblockcsi_controller_test.go

```diff
@@ -0,0 +1,182 @@
+				}
+
+				By("Getting ClusterRoleBinding")
+				crb := &rbacv1.ClusterRoleBinding{}
+				for _, clusterRoleBinding := range clusterRoleBindings {
+					Eventually(func() (*rbacv1.ClusterRoleBinding, error) {
+						err := k8sClient.Get(context.Background(),
+							testsutil.GetResourceKey(clusterRoleBinding, found.Name, ""), crb)
+						return crb, err
+					  }, timeout, interval).ShouldNot(BeNil())
+				}
+
+				By("Getting controller StatefulSet")
+				controller := &appsv1.StatefulSet{}
+				Eventually(func() (*appsv1.StatefulSet, error) {
+				  err := k8sClient.Get(context.Background(),
+				  	testsutil.GetResourceKey(config.CSIController, found.Name, found.Namespace), controller)
+				  return controller, err
+				}, timeout, interval).ShouldNot(BeNil())
+				checkContainersImages(controller.Spec.Template.Spec, containersImages)
+
+				By("Getting node DaemonSet")
+				node := &appsv1.DaemonSet{}
+				Eventually(func() (*appsv1.DaemonSet, error) {
+				  err := k8sClient.Get(context.Background(),
+				  	testsutil.GetResourceKey(config.CSINode, found.Name, found.Namespace), node)
+				  return node, err
+				}, timeout, interval).ShouldNot(BeNil())
+				checkContainersImages(node.Spec.Template.Spec, containersImages)
+
+				By("Checking if all containers were deployed")
+				var containersNameInControllerAndNode []string
+				containersNameInControllerAndNode = addContainersNameInPod(node.Spec.Template.Spec, containersNameInControllerAndNode)
+				containersNameInControllerAndNode = addContainersNameInPod(controller.Spec.Template.Spec, containersNameInControllerAndNode)
+				checkIfContainersInCrExistsInCSI(containersNameInControllerAndNode, containersImages)
+				
+				close(done)
+			  }, timeout.Seconds())
+		})
+	})
+})
+
+func checkContainersImages(podSpec corev1.PodSpec, containersImages map[string]string) {
+	Expect(podSpec.Containers).To(Not(BeEmpty()))
+	for _, container := range podSpec.Containers {
+		image, ok := containersImages[container.Name]
+		Expect(ok).To(BeTrue(), fmt.Sprintf("container %s not found in %s", container.Name, containersImages))
+		Expect(image).To(Equal(container.Image))
+	}
+}
+
+func addContainersNameInPod(podSpec corev1.PodSpec, containersNames []string) []string {
+	for _, container := range podSpec.Containers {
+		containersNames = append(containersNames, container.Name)
+	}
+	return containersNames
+}
+
+func checkIfContainersInCrExistsInCSI(containersNames []string, containersImages map[string]string) {
+	for containerName, _ := range containersImages{
+		Expect(isContainerDeployed(containersNames, containerName)).To(BeTrue(),
+			fmt.Sprintf("container %s not found in CSI deployment", containerName))
+	}
+}
+
+func isContainerDeployed(containersNames []string, wantedContainerName string) bool {
+    for _, containerName := range containersNames {
+        if containerName == wantedContainerName {
+            return true
+        }
+    }
+    return false
+}
```

Comment:
these are hard to miss when going over the diff
![image](https://user-images.githubusercontent.com/57443811/160292599-6374cc83-8a05-4081-88ef-34cd23ec3b19.png)
they are big and red... just remember to look for them, so you won't miss them this frequently

Reply:
done